airflow/sensors/time_delta.py

```diff
@@ -73,14 +73,10 @@ def execute(self, context: Context) -> bool | NoReturn:
             # If the target datetime is in the past, return immediately
             return True
         try:
-            trigger = DateTimeTrigger(moment=target_dttm)
+            trigger = DateTimeTrigger(moment=target_dttm, end_task=True)
         except (TypeError, ValueError) as e:
             if self.soft_fail:
                 raise AirflowSkipException("Skipping due to soft_fail is set to True.") from e
             raise
 
-        self.defer(trigger=trigger, method_name="execute_complete")
-
-    def execute_complete(self, context, event=None) -> None:
-        """Execute for when the trigger fires - return immediately."""
-        return None
+        self.defer(trigger=trigger, method_name=TaskDeferred.TRIGGER_EXIT)
```

Comment:
`TaskDeferred.TRIGGER_EXIT` looks like just a special value to indicate we want to end the task in the trigger. I'm wondering about whether we really need to make it part of the exception `TaskDerferred.` Just making it a dedicated class/lambda/variable/enum makes more sense to me.

Reply:
I have removed `TaskDeferred.TRIGGER_EXIT` as it was sentinel value to signal the expectation that the trigger will exit the task. If the trigger will terminate the task instance itself, method_name should be None. Otherwise, provide the name of the method to be used when resuming execution in the task.

Reply:
This value is irrelevant when the task is finished anyway.